### PR TITLE
feat: feature flag password guest link and improvements

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

1. add feature flag to prevent password link creation
2. fix: join dialog is not cancelable
3. fix: change shield icon tint to match the text color

https://github.com/wireapp/wire-android-reloaded/assets/9083456/edbc99c0-2695-4c87-b1e6-b255209d583d

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1247](https://10.10.124.134/job/AR-build-pipeline/1247/) **failed**.

**Review by alexandreferris — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [1248](https://10.10.124.134/job/AR-build-pipeline/1248/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.4.0(7390100)(7390100)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.4.0(7390100)(7390100)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1259](https://10.10.124.134/job/AR-build-pipeline/1259/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.4.0(7391324)(7391324)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.4.0(7391324)(7391324)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/WireActivity.kt [hunk 1]
```diff
@@ -282,9 +282,10 @@ class WireActivity : AppCompatActivity() {
             val onComplete: (convId: ConversationId?) -> Unit = remember {
                 {
                     it?.also {
-                        appLogger.d("Join conversation via code dialog completed, navigating to conversation screen")
                         navigate(NavigationCommand(ConversationScreenDestination(it), BackStackMode.CLEAR_TILL_START))
                         viewModel.onJoinConversationFlowCompleted()
+                    } ?: run {
+                        viewModel.onJoinConversationFlowCompleted()
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/WireActivity.kt]: what about calling it `onJoinConversationFlowCompleted` once after `it.also{}` ?

**Comment by AndroidBob** [comment]
Build [634](https://10.10.124.134/job/AR-build-pipeline/634/) **failed**.

**Comment by AndroidBob** [comment]
Build [641](https://10.10.124.134/job/AR-build-pipeline/641/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7514275)(7514275)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7514275)(7514275)-staging-compat.apk)

**Review by ohassine — APPROVED** [review]

**Pull request merged**